cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -44,7 +44,13 @@ import (
 	"golang.org/x/exp/maps"
 )
 
-const DefaultModule = "main"
+const (
+	DefaultModule = "main"
+
+	// Used to represent a nil value during environment variable substitution
+	// when the environment variable has not been defined
+	nilIdentifier = "${NIL}"
```

Comment:
nit.

```suggestion
	cNilIdentifier = "${NIL}"
```

Reply:
Lets discuss our constant naming convention.  Right now we're a bit all over the place and if we decide to have a `c` prefix that is fine.  Will wait till then.